barretenberg/cpp/src/barretenberg/vm/avm/trace/execution.cpp

```diff
@@ -444,10 +445,25 @@ AvmError Execution::execute_enqueued_call(AvmTraceBuilder& trace_builder,
         .da_gas_left = da_gas_allocated_to_enqueued_call,
         .internal_return_ptr_stack = {},
     };
-    trace_builder.allocate_gas_for_call(l2_gas_allocated_to_enqueued_call, da_gas_allocated_to_enqueued_call);
     // Find the bytecode based on contract address of the public call request
-    std::vector<uint8_t> bytecode =
-        trace_builder.get_bytecode(trace_builder.current_ext_call_ctx.contract_address, check_bytecode_membership);
+    std::vector<uint8_t> bytecode;
+    try {
+        bytecode =
+            trace_builder.get_bytecode(trace_builder.current_ext_call_ctx.contract_address, check_bytecode_membership);
+    } catch ([[maybe_unused]] const std::runtime_error& e) {
+        info("AVM enqueued call exceptionally halted. Error: No bytecode found for enqueued call");
+        // FIXME: properly handle case when bytecode is not found!
+        // For now, we add a dummy row in main trace to mutate later.
+        // Dummy row in main trace to mutate afterwards.
+        // This error was encountered before any opcodes were executed, but
+        // we need at least one row in the execution trace to then mutate and say "it halted and consumed all gas!"
+        trace_builder.op_add(0, 0, 0, 0, OpCode::ADD_8);
+        trace_builder.handle_exceptional_halt();
+        return AvmError::NO_BYTECODE_FOUND;
+        ;
+    }
+
+    trace_builder.allocate_gas_for_call(l2_gas_allocated_to_enqueued_call, da_gas_allocated_to_enqueued_call);
 
```

Comment:
i COULD have had `get_bytecode` return `[byteode, error]`... but this is way simpler...